packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,260 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {centerBaseline} from './CenterBaseline';
+import {ContextValue, DEFAULT_SLOT, Provider, TextContext as RACTextContext, Radio, RadioGroup, RadioGroupProps, RadioGroupStateContext, RadioProps} from 'react-aria-components';
+import {createContext, forwardRef, ReactNode, RefObject, useCallback, useContext, useRef} from 'react';
+import {DOMRef, DOMRefValue, FocusableRef} from '@react-types/shared';
+import {focusRing, getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import {IconContext} from './Icon';
+import {pressScale} from './pressScale';
+import {size, style} from '../style/spectrum-theme' with {type: 'macro'};
+import {Text, TextContext} from './Content';
+import {useDOMRef, useFocusableRef} from '@react-spectrum/utils';
+import {useLayoutEffect} from '@react-aria/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface SegmentedControlProps extends Omit<RadioGroupProps, 'isReadOnly' | 'name' | 'isRequired' | 'isInvalid' | 'validate' | 'validationBehavior' | 'children' | 'className' | 'style' | 'aria-label' | 'orientation'>, StyleProps{
+  /**
+   * The content to display in the segmented control.
+   */
+  children: ReactNode,
+  /**
+   * Whether the segmented control is disabled.
+   */
+  isDisabled?: boolean,
+  /**
+   * Defines a string value that labels the current element.
+   */
+  'aria-label': string
+}
+export interface SegmentedControlItemProps extends Omit<RadioProps, 'children' | 'className' | 'style' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, StyleProps {
+  /**
+   * The content to display in the control item.
+   */
+  children: ReactNode
+}
+
+export const SegmentedControlContext = createContext<ContextValue<SegmentedControlProps, DOMRefValue<HTMLDivElement>>>(null);
+
+const segmentedControl = style<{size: string}>({
+  font: 'control',
+  display: 'flex',
+  backgroundColor: 'gray-100',
+  borderRadius: 'lg',
+  width: 'full'
+}, getAllowedOverrides());
+
+const controlItem = style({
+  position: 'relative',
+  display: 'flex',
+  forcedColorAdjust: 'none',
+  color: {
+    default: 'gray-700',
+    isHovered: 'neutral-subdued',
+    isSelected: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'ButtonText',
+      isDisabled: 'GrayText',
+      isSelected: 'HighlightText'
+    }
+  },
+  // TODO: update this padding for icon-only items when we introduce the non-track style back
+  paddingX: {
+    default: 'edge-to-text',
+    ':has([slot=icon]:only-child)': size(6)
+  },
+  height: 32,
+  alignItems: 'center',
+  flexBasis: 0,
+  flexGrow: 1,
+  flexShrink: 0,
+  justifyContent: 'center',
+  whiteSpace: 'nowrap',
```

Comment:
should this have `truncate`? Tried using a real long label and got the following
<img width="581" alt="image" src="https://github.com/user-attachments/assets/29b6df5e-606f-49a0-be0a-6f8aa265adfa">